inst/include/hiv_demographic_projection.hpp

```diff
@@ -0,0 +1,168 @@
+#pragma once
+
+#include <iostream>
+#include "types.hpp"
+
+template <typename real_type>
+void run_hiv_pop_demographic_projection(int time_step,
+                                        const Parameters<real_type>& pars,
+                                        const State<real_type>& state_curr,
+                                        State<real_type>& state_next,
+                                        WorkingData<real_type>& working) {
+  run_hiv_ageing_and_mortality(time_step, pars, state_curr, state_next);
```

Comment:
So we have `run_ageing_and_mortality` and `run_hiv_ageing_and_mortality` - this really begs for some discussion of what is going on that is similar or different between these!